src/openms/source/ANALYSIS/ID/SimpleSearchEngineAlgorithm.cpp

```diff
@@ -0,0 +1,654 @@
+      double fragment_mass_tolerance,
+      const String& precursor_mass_tolerance_unit_ppm,
+      const String& fragment_mass_tolerance_unit_ppm,
+      const Int precursor_min_charge,
+      const Int precursor_max_charge,
+      const String& enzyme,
+      const String& database_name)
+  {
+    // remove all but top n scoring
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+    for (SignedSize scan_index = 0; scan_index < (SignedSize)annotated_hits.size(); ++scan_index)
+    {
+      // sort and keeps n best elements according to score
+      Size topn = top_hits > annotated_hits[scan_index].size() ? annotated_hits[scan_index].size() : top_hits;
+      std::partial_sort(annotated_hits[scan_index].begin(), annotated_hits[scan_index].begin() + topn, annotated_hits[scan_index].end(), AnnotatedHit_::hasBetterScore);
+      annotated_hits[scan_index].resize(topn);
+      annotated_hits.shrink_to_fit();
+    }
+
+#ifdef _OPENMP
+#pragma omp parallel for
+#endif
+    for (SignedSize scan_index = 0; scan_index < (SignedSize)annotated_hits.size(); ++scan_index)
+    {
+      if (!annotated_hits[scan_index].empty())
+      {
+        // create empty PeptideIdentification object and fill meta data
+        PeptideIdentification pi;
+        pi.setMetaValue("scan_index", static_cast<unsigned int>(scan_index));
+        pi.setScoreType("hyperscore");
+        pi.setHigherScoreBetter(true);
+        pi.setRT(exp[scan_index].getRT());
+        pi.setMZ(exp[scan_index].getPrecursors()[0].getMZ());
+        Size charge = exp[scan_index].getPrecursors()[0].getCharge();
+
+        // create full peptide hit structure from annotated hits
+        vector<PeptideHit> phs;
+        for (vector<AnnotatedHit_>::const_iterator a_it = annotated_hits[scan_index].begin(); a_it != annotated_hits[scan_index].end(); ++a_it)
+        {
+          PeptideHit ph;
+          ph.setCharge(charge);
+
+          // get unmodified string
+          AASequence aas = AASequence::fromString(a_it->sequence.getString());
+
+          // reapply modifications (because for memory reasons we only stored the index and recreation is fast)
+          vector<AASequence> all_modified_peptides;
+          ModifiedPeptideGenerator::applyFixedModifications(fixed_modifications.begin(), fixed_modifications.end(), aas);
+          ModifiedPeptideGenerator::applyVariableModifications(variable_modifications.begin(), variable_modifications.end(), aas, max_variable_mods_per_peptide, all_modified_peptides);
+
+          // reannotate much more memory heavy AASequence object
+          AASequence fixed_and_variable_modified_peptide = all_modified_peptides[a_it->peptide_mod_index]; 
+          ph.setScore(a_it->score);
+          ph.setSequence(fixed_and_variable_modified_peptide);
+          phs.push_back(ph);
+        }
+        pi.setHits(phs);
+        pi.assignRanks();
+
+#ifdef _OPENMP
+#pragma omp critical (peptide_ids_access)
+#endif
+        {
+          peptide_ids.push_back(pi);
```

Comment:
could you move them back here?
```suggestion
          peptide_ids.push_back(std::move(pi));
```